File: view/src/lib/types/Duration.ts
export enum Duration {
  Instantaneous = "Instantaneous",
  OneRound = "1 round",
  OneMinute = "1 minute",
  TenMinutes = "10 minutes",
  OneHour = "1 hour",
  EightHours = "8 hours",
  OneDay = "24 hours",
  SevenDays = "7 days",
  Permanent = "Permanent",
  UntilDispelled = "Until dispelled",
}

export const Durations: Duration[] = Object.values(Duration);

File: crates/types/src/spell/duration.rs
use strum::{Display, VariantArray};

#[derive(Debug, Display, VariantArray, Clone, Copy, PartialEq)]
pub enum SpellDuration {
    Instantaneous,
    #[strum(to_string = "1 round")]
    OneRound,
    #[strum(to_string = "1 minute")]
    OneMinute,
    #[strum(to_string = "10 minutes")]
    TenMinutes,
    #[strum(to_string = "1 hour")]
    OneHour,
    #[strum(to_string = "8 hours")]
    EightHours,
    #[strum(to_string = "1 day")]
    OneDay,
    #[strum(to_string = "7 days")]
    SevenDays,
    Permanent,
    #[strum(to_string = "Until dispelled")]
    UntilDispelled,
}
